app/models/search/collection_indexer.rb

```diff
@@ -0,0 +1,102 @@
+class CollectionIndexer < Indexer
+  def self.klass
+    "Collection"
+  end
+
+  def self.mapping
+    {
+      "collection" => {
+        properties: {
+          title: {
+            type: "text",
+            analyzer: "collection_title_analyzer",
+            fields: {
+              exact: {
+                type:     "text",
+                analyzer: "exact_collection_analyzer"
+              },
+              keyword: {
+                type: "keyword",
+                normalizer: "keyword_lowercase"
+              }
+            }
+          },
+          challenge_type: {            
+            type: "keyword",
+            null_value: "NULL" 
+          },
+          name: { type: "text", analyzer: "simple" },
+          description: { type: "text", analyzer: "simple" },
+          collection_type: { type: "keyword" },
+          created_at: { type: "date" }
+        }
+      }
+    }
+  end
+
+  def self.settings
+    {
+      analysis: {
+        analyzer: {
+          collection_title_analyzer: {
+            type: "custom",
+            tokenizer: "standard",
+            filter: [
+              "lowercase"
+            ]
+          },
+          exact_collection_analyzer: {
+            type: "custom",
+            tokenizer: "keyword",
+            filter: [
+              "lowercase"
+            ]
+          }
+        },
+        normalizer: {
+          keyword_lowercase: {
+            type: "custom",
+            filter: ["lowercase"]
+          }
+        }
+      }
+    }
+  end
+
+  def document(object)
+    object.as_json(
+      root: false,
+      only: [
+        :id, :name, :title, :description, :parent_id, :collection_type, :created_at
+      ]
+    ).merge(
+      closed: object.closed?,
+      unrevealed: object.unrevealed?,
+      anonymous: object.anonymous?,
+      owner_ids: object.all_owners.pluck(:user_id),
+      moderated: object.moderated?,
+      moderator_ids: object.all_moderators.pluck(:user_id),
+      maintainer_ids: object.maintainers.pluck(:user_id),
+      challenge_type: object.challenge_type,
+      signup_open: object.challenge&.signup_open,
+      signups_open_at: object.challenge&.signups_open_at,
+      signups_close_at: object.challenge&.signups_close_at,
+      assignments_due_at: object.challenge&.assignments_due_at,
+      works_reveal_at: object.challenge&.works_reveal_at,
+      authors_reveal_at: object.challenge&.authors_reveal_at,
+      general_works_count: object.all_approved_works.count,
+      public_works_count: object.all_approved_works.where(restricted: false).count,
+      general_bookmarked_items_count: get_bookmarked_items_count(object), 
+      public_bookmarked_items_count: get_bookmarked_items_count(object, true)
```

Comment:
[ticking had a comment here](https://github.com/otwcode/otwarchive/pull/3912#discussion_r528200411) that got buried:

>Does this update properly when a work or series with a bookmark in the collection changes from restricted to unrestricted? What about when a bookmark changes visibility? (Either private or hidden_by_admin.) What about when the bookmark is in a subcollection, for either case?

I think changes to _bookmark_ visibility are covered in the bookmark model, although as I noted in a comment on `Bookmark#update_pseud_and_collection_index`, it might need a tweak to handle when the bookmark is in a subcollection.

However, I don't see where reindexing is being handled for _bookmarkables_ (works, series, external works) that change visibility. I'm pretty sure works, series, and external works can all have their `hidden_by_admin` value changed, and works or series can have their `restricted` value changed.